radar-upload-backend/src/main/java/org/radarbase/upload/resource/RecordResource.kt

```diff
@@ -152,11 +150,60 @@ class RecordResource {
         return recordMapper.fromRecords(recordRepository.poll(imposedLimit), imposedLimit)
     }
 
+    @GET
+    @Path("{recordId}/metadata")
+    fun getRecordMetaData(@PathParam("recordId") recordId: Long): RecordMetadataDTO {
+
+        val record = recordRepository.read(recordId)
+                ?: throw NotFoundException("Record with ID $recordId does not exist")
+
+        return recordMapper.fromMetadata(record.metadata)
+    }
+
+    @POST
+    @Path("{recordId}/metadata")
+    fun updateRecordMetaData(metaData: RecordMetadataDTO, @PathParam("recordId") recordId: Long): RecordMetadataDTO {
+
+        val record = recordRepository.read(recordId)
+                ?: throw NotFoundException("Record with ID $recordId does not exist")
+
+        record.metadata.canBeUpdatedTo(metaData)
+
+        val updatedRecord = recordRepository.update(recordMapper.toMetadata(metaData, record.metadata))
+
+        return recordMapper.fromMetadata(updatedRecord)
+    }
+
+    private fun RecordMetadata.canBeUpdatedTo(metaDataToUpdate: RecordMetadataDTO) {
```

Comment:
We will need to move this logic to the RecordRepository inside a transaction, so that the condition check and actual update happen atomically. Otherwise you can still get two concurrent converters that modify the metadata simultaneously.